sympy/physics/control/lti.py

```diff
@@ -1226,224 +1179,909 @@ def get_asymptotic_stability_conditions(self, cancel_poles_zeros = False):
+    def __init__(self, num, den, var, sampling_time = 1):
+        super().__init__()
+        self._sampling_time = sampling_time
 
-    __rmul__ = __mul__
+    @classmethod
+    def from_rational_expression(cls, expr, var=None, sampling_time = 1):
+        r"""
+        See :func:`TransferFunctionBase.from_rational_expression`.
+        """
+        return super().from_rational_expression(expr, var,
+                                         sampling_time = sampling_time)
 
-    def __truediv__(self, other):
-        if isinstance(other, TransferFunction):
-            if not self.var == other.var:
-                raise ValueError(filldedent("""
-                    All the transfer functions should use the same complex variable
-                    of the Laplace transform."""))
-            return Series(self, TransferFunction(other.den, other.num, self.var))
-        elif (isinstance(other, Parallel) and len(other.args
-                ) == 2 and isinstance(other.args[0], TransferFunction)
-            and isinstance(other.args[1], (Series, TransferFunction))):
+    @classmethod
+    def from_coeff_lists(cls, num_list, den_list, var, sampling_time = 1):
+        r"""
+        See :func:`TransferFunctionBase.from_coeff_lists`.
+        """
+        return super().from_coeff_lists(num_list, den_list,
+                                        var, sampling_time = sampling_time)
 
-            if not self.var == other.var:
-                raise ValueError(filldedent("""
-                    Both TransferFunction and Parallel should use the
-                    same complex variable of the Laplace transform."""))
-            if other.args[1] == self:
-                # plant and controller with unit feedback.
-                return Feedback(self, other.args[0])
-            other_arg_list = list(other.args[1].args) if isinstance(
-                other.args[1], Series) else other.args[1]
-            if other_arg_list == other.args[1]:
-                return Feedback(self, other_arg_list)
-            elif self in other_arg_list:
-                other_arg_list.remove(self)
-            else:
-                return Feedback(self, Series(*other_arg_list))
+    @classmethod
+    def from_zpk(cls, zeros, poles, gain, var, sampling_time = 1):
+        r"""
+        See :func:`TransferFunctionBase.from_zpk`.
+        """
+        return super().from_zpk(zeros, poles, gain,
+                                var, sampling_time = sampling_time)
 
-            if len(other_arg_list) == 1:
-                return Feedback(self, *other_arg_list)
-            else:
-                return Feedback(self, Series(*other_arg_list))
-        else:
-            raise ValueError("TransferFunction cannot be divided by {}.".
-                format(type(other)))
+    @classmethod
+    def gbt(cls, cont_tf, sampling_time, alpha, var):
+        r"""
+        Returns the discretized transfer function H(z) from a continuous
+        transfer function H(s).
```

Comment:
For methods with args/kwargs besides self, make sure to have a parameters section.